server/src/main/java/org/opensearch/plugins/ResourcePlugin.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+/**
+ * This interface should be implemented by all the plugins that define one or more resources.
+ *
+ * @opensearch.experimental
+ */
+public interface ResourcePlugin {
+
+    /**
+     * Type of the resource
+     * @return a string containing the type of the resource
+     */
+    String getResourceType();
+
+    /**
+     * The index where resource meta-data is stored
+     * @return the name of the parent index where resource meta-data is stored
+     */
+    String getResourceIndex();
```

Comment:
Can a plugin define multiple types of resources that can be shared?

i.e. In a hypothetical Searchable Photos plugin, would it be able to do resource sharing on both the Album level and Photo level?

Reply:
Yes. Since the index can hold only one type of record, plugins defining different types of resources must define an index corresponding to that resource.

Reply:
I think this should be more flexible in case a plugin defines multiple resource types. Maybe make a sub-interface and under the ResourcePlugin interface return a List of the subtype?

i.e.

```
public interface ResourcePlugin {
    List<ResourceType> getResourceTypes();

   // assign a ResourceService per resource type that can be interrogated for resource access queries.
  void assignResourceService(List<ResourceService<T>> resourceService...)
}

public interface ResourceType {

    /**
     * Type of the resource
     * @return a string containing the type of the resource
     */
    String getResourceType();

    /**
     * The index where resource meta-data is stored
     * @return the name of the parent index where resource meta-data is stored
     */
    String getResourceIndex();
}
```